pkg/controller/appdefinition/status.go

```diff
@@ -588,3 +588,25 @@ func endpoints(req router.Request, cfg *apiv1.Config, app *v1.AppInstance) (stri
 
 	return strings.Join(endpointStrings, ", "), nil
 }
+
+func RefreshStatus(h router.Handler) router.Handler {
+	return router.HandlerFunc(func(req router.Request, resp router.Response) error {
+		appInstance := req.Object.(*v1.AppInstance)
+
+		if appInstance.Annotations["acorn.io/refresh"] == "true" {
+			delete(appInstance.Annotations, "acorn.io/refresh")
+			for i, annotation := range appInstance.Spec.Annotations {
+				if annotation.Key == "acorn.io/refresh" {
+					appInstance.Spec.Annotations = append(appInstance.Spec.Annotations[:i], appInstance.Spec.Annotations[i+1:]...)
```

Comment:
This likely introduces a weird behavior if appInstance.Spec.Annotations happens to have acorn.io/refresh=true two or more times. I think it would never be able to get rid of its label, and keeps almost doubling the amount of labels.

Reply:
good catch, I modified to start from then end and remove which I think resolves this

Reply:
added a test case for this as well